core/src/test/java/org/hibernate/ogm/backendtck/elementcollection/DuplicatesAndOrderOfAssociationItemsTest.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Otherwise that will not guarantee. It will depend on the specific datastore behavior.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+@TestForIssue( jiraKey = "OGM-1237" )
+public class DuplicatesAndOrderOfAssociationItemsTest extends OgmTestCase {
+
+	public static final String NAME = "Programmers F2F meeting";
+	public static final String ADDRESS = "1 avenue des Champs Elysees, Paris, France";
+	public static final String ADDRESS_ALT1 = "Piazza del Colosseo, 1, Rome, Italy";
+	public static final String ADDRESS_ALT2 = "Charing Cross, London WC2N 5DU, UK";
+	public static final String PHONE = "+1-202-555-0333";
+	public static final String PHONE_ALT1 = "+1-202-555-0333";
+	public static final String PHONE_ALT2 = "+1-202-555-0333";
+	public static final String PARTICIPANT_0 = "telegraph4eternity";
+	public static final String PARTICIPANT_1 = "fax4ever";
+	public static final String PARTICIPANT_2 = "phone4induction";
+	public static final String SPEAKER = "unclebob";
+
+	@After
+	public void tearDown() {
+		deleteAll( Programmer.class, PARTICIPANT_0, PARTICIPANT_1, PARTICIPANT_2, SPEAKER );
+		deleteAll( Meeting.class, NAME );
+	}
+
+	@Test
+	public void testKeepDuplicatesOnAssociationItems() {
+		Meeting meeting = new Meeting();
+		meeting.setName( NAME );
+
+		meeting.addAddress( ADDRESS );
+		meeting.addAddress( ADDRESS_ALT1 );
+		meeting.addAddress( ADDRESS_ALT2 );
+
+		meeting.addPhone( PHONE );
+		meeting.addPhone( PHONE );
+		meeting.addPhone( PHONE );
+
+		Programmer participant0 = new Programmer( PARTICIPANT_0 );
+		Programmer participant1 = new Programmer( PARTICIPANT_1 );
+		Programmer participant2 = new Programmer( PARTICIPANT_2 );
+
+		meeting.addParticipant( participant0 );
+		meeting.addParticipant( participant1 );
+		meeting.addParticipant( participant2 );
+
+		Programmer speaker = new Programmer( SPEAKER );
+		meeting.addSpeaker( speaker );
+		meeting.addSpeaker( speaker );
+		meeting.addSpeaker( speaker );
+
+		inTransaction( session -> {
+			session.save( participant0 );
+			session.save( participant1 );
+			session.save( participant2 );
+			session.save( speaker );
+			session.save( meeting );
+		} );
+
+		inTransaction( session -> {
+			Meeting load = session.load( Meeting.class, NAME );
+
+			// an embedded list with a defined @OrderColumn keeps duplicates
+			assertThat( load.getPhones() ).containsExactly( PHONE, PHONE, PHONE );
+
+			// without an @OrderCoulmn it may or may not keep duplicate
+			assertThat( load.getAddresses() ).containsOnly( ADDRESS, ADDRESS_ALT1, ADDRESS_ALT2 );
+
+			// a join table list with a defined @OrderColumn keeps duplicates
+			assertThat( load.getSpeakers() ).containsExactly( speaker, speaker, speaker );
+
+			// without an @OrderCoulmn it may or may not not keep duplicate
+			assertThat( load.getParticipants() ).containsOnly( participant0, participant1, participant2 );
+		} );
+	}
+
+	@Test
+	public void testPreserveOrderOnAssociationItems() {
```

Comment:
Isn't this test the same as List#testOrderedList

Reply:
Let's keep it anyway, but let's be careful not too have too many similar tests

Reply:
We verify the order here, instead of the duplicate keeping.

Reply:
> We verify the order here, instead of the duplicate keeping.

 The same thing we do in `List#testOrderedList`.
Also this issue was about testing duplicates and there are no duplicates here.

Just saying, this class become much shorter and simpler if we only focus on what we are working on without loosing in test coverage

Reply:
I think it isn't the same. In the latter we check order.

Reply:
Are you saying we are not checking the order of the elements here? https://github.com/hibernate/hibernate-ogm/blob/2a7c097408f6b8fcc771cd5bee570b8c588eed5d/core/src/test/java/org/hibernate/ogm/backendtck/associations/collection/types/ListTest.java#L62

Reply:
Whether I try to substitute:

> assertThat( load.getAddresses() ).containsOnly( ADDRESS, ADDRESS_ALT1, ADDRESS_ALT2 );

with:

> assertThat( load.getAddresses() ).containsExactly( ADDRESS, ADDRESS_ALT1, ADDRESS_ALT2 );

I got this:

> org.junit.ComparisonFailure: 
Expected :['1 avenue des Champs Elysees, Paris, France', 'Piazza del Colosseo, 1, Rome, Italy', 'Charing Cross, London WC2N 5DU, UK']
Actual   :['Charing Cross, London WC2N 5DU, UK', '1 avenue des Champs Elysees, Paris, France', 'Piazza del Colosseo, 1, Rome, Italy']